kythe/go/serving/tools/http_server/BUILD

```diff
@@ -20,3 +21,25 @@ go_binary(
         "@org_golang_x_net//http2:go_default_library",
     ],
 )
+
+go_image(
```

Comment:
What's the context/reasoning/need for this target?

Reply:
[`go_image`](https://github.com/bazelbuild/rules_docker#go_image) is a `rules_docker` [macro](https://github.com/bazelbuild/rules_docker/blob/master/go/image.bzl#L65) that does two things.

1. It creates a `rules_go` `go_binary` named `name + ".binary"`.
1. Creates a docker image using that binary.

If you `blaze run` this target on a machine with Docker installed it will create a new tagged docker image named `bazel/kythe/go/serving/tools/http_server/http_server_image:http_server_image` (name:tag). This docker image is constructed entirely hermetically in Bazel without invoking the Docker tools which means you don't need `dind` in order to build an image. By default the `go_image` rule uses the `grc.io/distroless/base` as the base for the binaries. Interestingly the `gcr.io/distroless/base` image is itself created in Bazel this way.

In what way would you like me to expand in the PR to make this more clear? Would you prefer this as a change to the Kythe documentation, to my commit message, or inline in the `BUILD` file itself?

Reply:
I would like for more of the motivation for this target to be added to the description.  While I understand that this creates a Docker image, it's not clear at all why we would want this image since the Kythe team does not need it.

Reply:
How does the Kythe team serve their UI if not through the sample web UI? Is there a more fleshed out example that can be used? Is the expectation that a user of Kythe will fork off the sample UI and write their own server?

Reply:
I don't follow.  A Docker image is not necessary to run the server.  The lack of a Docker image solely for this binary does not imply an expectation to fork the server code.